ospf6d/ospf6_message.c

```diff
@@ -1684,91 +1830,243 @@ int ospf6_receive(struct thread *thread)
 		assert(0);
 	}
 
+	return OSPF6_READ_CONTINUE;
+}
+
+int ospf6_receive(struct thread *thread)
+{
+	int sockfd;
+	struct ospf6 *ospf6;
+	int count = 0;
+
+	/* add next read thread */
+	ospf6 = THREAD_ARG(thread);
+	sockfd = THREAD_FD(thread);
+
+	thread_add_read(master, ospf6_receive, ospf6, ospf6->fd,
```

Comment:
Since we're run to completion I suppose it doesn't really matter where it is queued as long as there is no chance of returning rarly and failing to do it. Happy to move this if it makes more sense.

Reply:
actually - there is a return on error so It is better to have the thread read scheduled before to give the socket a chance to receive again after a transient error.

Reply:
I don't see a return on error before 1858. But it's fine if you want to keep it this way.